### PR TITLE
Add open API to allow the creation of emission from another APP

### PR DESCRIPTION
**Pull request by Jeconias**
nmf-openApi

Don't forget to add `{
  "expo": {
    "scheme": "nmfearth"
  }
}` in app.json.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "expo-asset": "8.2.0",
     "expo-constants": "9.2.0",
     "expo-font": "8.3.0",
+    "expo-linking": "^1.0.4",
```
Comment (PierreBresson) [inline package.json:43]: no ^

**Review by PierreBresson — CHANGES_REQUESTED** [review]
Nice pr, clean code but this system can't work. It's partially my fault, I should have given more guidance sorry. It's because if, let's say Air France, wants to give its user the ability to add their precise flight emission in NMF.earth app this will not work.

They will have a better calculation of the co2 impact (since they have access to airplane data) and it will differ from our calculation.

In order to allow external apps to put exact amount of co2, we need to introduced a way to bypass our calculation. To do so, you can add an optional property to emission in and in `emission.interface.ts` :

```
interface EmissionPayload {
  isValueCO2Value?: boolean;
...
```

and in `calculation.ts` : 

```
const getC02ValueFromEmission = (emission: Emission) => {
  if (emission.isValueCO2Value) {
    return emission.value;
  }
```

Therefore the linking will not open `Add Emission` screen but `Emissions` instead and ask the user with a simple [alert](https://reactnative.dev/docs/alert) to confirm if they want to add this emission. Behind the scenes, the emission should be simply added to the redux store.

Don't forget to add to readme some explanation for external devs on how to use this api (and also for me to check your pr).

**Comment by Jeconias** [comment]
OK, I will work on these changes.

**Comment by PierreBresson** [comment]
@Jeconias  Due to #243 , `master` branch is now deleted and `main` is the new `master`, that's why the PR was closed and I can't reopen it, sorry :/

**Comment by Jeconias** [comment]
ok @PierreBresson! I had some problems and I couldn't finish quickly. I will rebase and finish in the next few days.

**Pull request closed without merging**